src/engine/TPA.cc

```diff
@@ -1353,152 +1342,211 @@ class TransitionSystemNetworkManager {
     NetworkNode & getNode(SymRef vid) { return networkMap.at(vid); }
     NetworkNode const & getNode(SymRef vid) const { return networkMap.at(vid); }
 
-    SymRef getChainStart() const { return graph.getTarget(adjacencyRepresentation.getOutgoingEdgesFor(graph.getEntry())[0]); }
+    vec<SymRef> getGraphStarts() const {
+        vec<SymRef> graphStarts;
+        for (uint i = 0; i < adjacencyRepresentation.getOutgoingEdgesFor(graph.getEntry()).size(); i++) {
+            graphStarts.push(graph.getTarget(adjacencyRepresentation.getOutgoingEdgesFor(graph.getEntry())[i]));
+        }
```

Comment:
Store return value for `adjacencyRepresentation.getOutgoingEdgesFor` before the loop. Use range-based for loop afterwards.

Reply:
Fixed